### PR TITLE
Add postman collection and README for general workflow

### PR DESCRIPTION
**Pull request by mallek**
I added the postman collection we used at ETHDenver to the /examples/postman and added a quick readme to describe the workflow

### CODE REVIEW

**Comment by mallek** [comment]
I see a heartbeat_demo test is failing, would you like to fix that and I can merge it into this PR?

**Comment by mswilkison** [comment]
Hey @mallek and team - thanks for this! We're working on merging in some other pull requests but will try to give this our attention in the next several days.

File: examples/postman/dev.postman_environment.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		{
+			"key": "alice_treasure_map",
+			"value": "",
+			"enabled": true
+		}
+	],
+	"_postman_variable_scope": "environment",
+	"_postman_exported_at": "2019-02-18T16:37:46.501Z",
+	"_postman_exported_using": "Postman/6.5.2"
+}
```
Comment (KPrasch) [inline examples/postman/dev.postman_environment.json]: missing newline at EOF

**Comment by KPrasch** [comment]
Awesome! leaving this open for a bit as we investigate how to include the changeset in CI.

File: examples/postman/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+The order to call the collection is as follows
+
+- Alice/derive_policy_pubkey (create a policy)
+- Enrico/encrypt_message (encrypt some data inside the policy)
+- Bob/public_keys (Get Bob's keys so alice can grant him permission)
+- Alice/grant (Grant bob permission to see the policy's data)
+- Bob/retrieve (Get alice's data)
+
+If you try to recieve the data before you are granted access you will be returned a 500 in the current version 0.0.14-alpha
```
Comment (cygnusv) [inline examples/postman/README.md]:
```suggestion
If you try to receive the data before you are granted access you will be returned a 500.
```

**Comment by mallek** [comment]
I had also suggested a tool like newman to run the tests.  I like making the calls because it allows the http context to be generated without any mocking.  Because of that his will also be a good place to test things like CORS access or TLS if that is intended to be added to the product. 

I also think having a postman collection may be easy for a first time user to pick up without having to read and understand the python tests.

I do want to make sure the documentation stays up to date.  I think this would be done if it is truly used by the community where if it's not used it will be neglected.

**Comment by KPrasch** [comment]
I actually do use postman, and I think this is a nice step for lowering the bar to entry; However, our experience tells us that anything that is untested will eventually fail.  

I'm looking forward to having this contribution merged if we can find a way to include in in CI.

File: examples/postman/README.md (hunk 2 above)
Comment (KPrasch) [reply examples/postman/README.md]: version number is wrong here - it is `v0.1.0-alpha.14`

Comment (cygnusv) [reply examples/postman/README.md]: I think it's better to not mention a specific version. An in any case, if we leave that like that, wouldn't bumpversion update it automatically, defeating this way the original intention of the sentence?

Comment (mallek) [reply examples/postman/README.md]: I can remove the version, it was more to document the return, If we add this to CI the test will say we expect a 500 for the first call and no version will be necessary.  we could assert a version is returned if wanted.

Comment (KPrasch) [reply examples/postman/README.md]: :+1:

**Comment by mallek** [comment]
We have identified that newman would be an appropriate way to run these tests. What additional work would you like to see done on this PR?

**Comment by cygnusv** [comment]
@mallek can you prepare a python script that runs it? We can use that as a basic for tests integrated in our CI

**Comment by mallek** [comment]
@cygnusv. Newman is a command line tool.  You want me to use python to start the cli tool?  The way I have done this is run it as a step in CI shelling out to run Newman.

**Comment by cygnusv** [comment]
No need to use python to run the CLI, I think. Just specify what are the newman commands we need to run, and we can integrate them in the CI workflow.

**Comment by mallek** [comment]
Is there a set of running actors available to hit via http during the CI process?

**Comment by cygnusv** [comment]
We prefer to deploy them locally in the test workflow. See this test as an example, which deploys a fleet or Ursulas to test the heartbeat demo

https://github.com/nucypher/nucypher/blob/ec6b754ad86e7ee504b8bb1f6031c9f3f11e98af/.circleci/config.yml#L444

File: examples/postman/NuCypher.postman_collection.json [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+					"port": "{{bob_port}}",
+					"path": [
+						"retrieve"
+					]
+				}
+			},
+			"response": []
+		}
+	]
+}
```
Comment (KPrasch) [inline examples/postman/NuCypher.postman_collection.json]: Missing Newline at OEF

**Comment by mallek** [comment]
I rebased this PR from #820 and made the changes requested

**Comment by KPrasch** [comment]
Heh - looks like you picked up an extra 172 commits there! (perhaps due to diverged history.)

**Comment by mallek** [comment]
Interesting I show a clean rebase from master.  I haven't worked much with upstream only origin so i'll play around with it and see if i can get it.  If not since it's just documentation I can just create a new PR from master

**Comment by KPrasch** [comment]
Be sure to add `nucypher/nucypher` as an upstream remote, and rebase over `upstream/master` after a fresh `git remote update`

**Comment by mallek** [comment]
that was it, I rebased over origin/master I'll try it... thanks

**Comment by mallek** [comment]
This should be much better, sorry :|

**Comment by KPrasch** [comment]
No worries! :-)

**Comment by mallek** [comment]
something in my tooling keeps removing the new line..

**Comment by KPrasch** [comment]
Issue #840

File: examples/postman/README.md (hunk 2 above)
Comment (tuxxy) [reply examples/postman/README.md]: @mallek Can you apply this suggestion to your PR here? We'll merge it as soon as that is in.

**Comment by mallek** [comment]
@tuxxy @KPrasch I understand that the automated tests need to be integrated to accept this PR, I however do not feel comfortable coding this into the CI.  If someone wants to take the ball and do that great if not I can close this PR.

**Pull request merged**